src/main/exit-app.ts

```diff
@@ -0,0 +1,18 @@
+import { app } from "electron";
+import { WindowManager } from "./window-manager";
+import { appEventBus } from "../common/event-bus";
+import { ClusterManager } from "./cluster-manager";
+import logger from "./logger";
+
+
+export function exitApp() {
+  const windowManager = WindowManager.getInstance<WindowManager>()
+  const clusterManager = ClusterManager.getInstance<ClusterManager>()
+  appEventBus.emit({ name: "service", action: "close" })
+  windowManager?.hide();
+  clusterManager?.stop();
```

Comment:
These are guaranteed to exist, the `getInstance` function creates if not present.

Reply:
Good point.